keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -70,6 +74,12 @@ class SqlArtifactRepository(
   private val sqlRetry: SqlRetry,
   private val artifactSuppliers: List<ArtifactSupplier<*, *>> = emptyList()
 ) : ArtifactRepository {
+
+  companion object {
+    private val ARTIFACT_VERSIONS_BRANCH =
+      field<String?>("keel.artifact_versions.git_metadata->'$.branch'")
```

Comment:
@robfletcher I know we ran into performance issues with JSON queries before. Should we look into creating a new view for artifacts like we did for resources?

Reply:
What area of performance are you concerned about?

Reply:
The last time I added a query for a JSON field, it ended up slowing queries down, which @robfletcher caught and corrected here: https://github.com/spinnaker/keel/pull/1378

This is why I pinged him. I should have looked for that PR earlier. My specific question is: should I add a generated column and an index, like we did for the `events` table, as part of this PR (or a follow-up)?

Reply:
I think these could be good candidates for a new column in the `artifact_versions` table. It seems like we might be querying for those things, so they should probably be in their own separate field. I do think that adding the JSON field here will slow things down

Reply:
Sounds good. I'll add the generated columns in a follow-up PR.